ViewModels/RegisterViewModel.cs

```diff
@@ -0,0 +1,137 @@
+            set
+            {
+                if (_account != value)
+                {
+                    _account = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        private ErrorMessage _error = new ErrorMessage();
+
+        public ErrorMessage Error
+        {
+            get => _error;
+            set
+            {
+                if (_error != value)
+                {
+                    _error = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        private bool _isPassword = true;
+        public bool IsPassword
+        {
+            get => _isPassword;
+            set
+            {
+                if (_isPassword != value)
+                {
+                    _isPassword = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        private bool _isPassword2 = true;
+        public bool IsPassword2
+        {
+            get => _isPassword2;
+            set
+            {
+                if (_isPassword2 != value)
+                {
+                    _isPassword2 = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public ICommand RegisterCommand { get; }
+        public ICommand GoToLoginCommand { get; }
+        public ICommand TogglePasswordVisibilityCommand { get; }
+        public ICommand TogglePasswordRepeatVisibilityCommand { get; }
+
+
+        public RegisterViewModel(INavigationService navigationService)
+        {
+            _navigationService = navigationService;
+
+            _validator = new RegistrationValidator();
+
+            RegisterCommand = new Command(OnRegistration);
+            GoToLoginCommand = new Command(OnGoToLogin);
+            TogglePasswordVisibilityCommand = new Command(OnTogglePasswordVisibility);
+            TogglePasswordRepeatVisibilityCommand = new Command(OnTogglePasswordRepeatVisibility);
+
+        }
+
+        private async void OnRegistration()
+        {
+            ValidationResult result = await _validator.ValidateAsync(Account);
+
+            if (!result.IsValid)
+            {
+                Error.Message = string.Join("\n", result.Errors.Select(e => e.ErrorMessage));
+                Error.IsVisible = true;
+                OnPropertyChanged(nameof(Error));
+                return;
+            }
+
+            if (!Account.Password.Equals(Account.PasswordRepeat)) 
+            {
+                Error.Message = "Repeat password";
+                Error.IsVisible = true;
+                OnPropertyChanged(nameof(Error));
+                return;
+            }
+
+            await _navigationService.NavigateToAsync("Home", null);
+        }
+
+        private async void OnGoToLogin()
+        {
+            await _navigationService.NavigateToAsync("Login", null);
+        }
+
+        private void OnTogglePasswordVisibility()
```

Comment:
Since this code is the same (and will most of the time be the same) for all viewmodels, it can be moved to a base class and the viewmodels can inherit from it. And you can make methods virtual in case you need to override them

so move the INotifyPropertyChanged related code to a base class
also just fyi different libraries are also often used for implementing this